tests/functions/test_groupby_topk.py

```diff
@@ -0,0 +1,67 @@
+import pandas as pd
+import pytest
+from pandas.testing import assert_frame_equal
+
+
+@pytest.fixture
+def df():
+    return pd.DataFrame(
+        [
+            {"age": 22, "major": "science", "ID": 145, "result": "pass"},
+            {"age": 20, "major": "history", "ID": 322, "result": "fail"},
+            {"age": 23, "major": "history", "ID": 321, "result": "pass"},
+            {"age": 21, "major": "law", "ID": 49, "result": "fail"},
+            {"age": 19, "major": "mathematics", "ID": 224, "result": "pass"},
+            {"age": 20, "major": "science", "ID": 132, "result": "pass"},
+        ]
+    )
+
+
+def test_ascending_groupby_k_2(df):
+    """Test ascending group by, k=2"""
+    expected = df.groupby("result").apply(
+        lambda d: d.sort_values("age").head(2)
+    )
+    assert_frame_equal(df.groupby_topk("result", "age", 2), expected)
+
+
+def test_descending_groupby_k_3(df):
+    """Test descending group by, k=3"""
+    expected = df.groupby("result").apply(
+        lambda d: d.sort_values("age", ascending=False).head(3)
+    )
+    assert_frame_equal(
+        df.groupby_topk("result", "age", 3, {"ascending": False}), expected
+    )
+
+
+def test_wrong_groupby_column_name(df):
+    """Raise Value Error if wrong groupby column name is provided."""
+    with pytest.raises(ValueError):
+        df.groupby_topk("RESULT", "age", 3)
+
+
+def test_wrong_sort_column_name(df):
```

Comment:
This also should be a KeyError exception, if we allow Pandas to do the checks

Reply:
Changed to KeyError.